### PR TITLE
RAC 5099 Implement UCS redfish Api GET /Systems/{identifier}/SimpleStorage

### PR DESCRIPTION
**Pull request by mcgG**
**Background**
----
UCS redfish is broke GET /Systems/{identifier}/SimpleStorage

----

Details
----
1. Implemented /Systems/{identifier}/SimpleStorage API to get Dell/Cisco/Other simple storage controllers.

@iceiilin  @anhou @pengz1 @yaolingling

### CODE REVIEW

File: lib/api/redfish-1.0/systems.js [hunk 1]
```diff
@@ -714,25 +715,43 @@ var listSimpleStorage = controller(function(req, res)  {
...
-                _.forOwn(controllers, function(val, key) { ids.push(key); });
-                options.controllers = ids;
+                return redfish.render('redfish.1.0.0.simplestoragecollection.json',
+                                'SimpleStorageCollection.json#/definitions/SimpleStorageCollection',
+                                options);
+            }).catch(function(error) {
+                return redfish.handleError(error, res);
+            });
+        } else if(result.vendor === 'Cisco') {
+            return Promise.resolve(dataFactory(identifier, 'UCS:board'))
```
Comment (pengz1) [inline lib/api/redfish-1.0/systems.js]: Promise.resolve is not necessary

File: lib/api/redfish-1.0/systems.js [hunk 2]
```diff
@@ -714,25 +715,43 @@ var listSimpleStorage = controller(function(req, res)  {
...
+            .then(function(storage) {
+                var controllers = [];
+                var sasControllers = storage.data.children['storage-SAS-collection'];
+                var flexControllers = storage.data.children['storage-FlexFlash-collection'];
+                var pchControllers = storage.data.children['storage-PCH-collection'];
```
Comment (pengz1) [inline lib/api/redfish-1.0/systems.js]: We don't know how many kinds of storage collection there are in UCS system. It is not a good idea to use such kind of enumerating methods.

File: lib/api/redfish-1.0/systems.js [hunk 3]
```diff
@@ -714,25 +715,44 @@ var listSimpleStorage = controller(function(req, res)  {
...
+        } else if(result.vendor === 'Cisco') {
+            return Promise.resolve(dataFactory(identifier, 'UCS:board'))
+            .then(function(storage) {
+                var controllers = [];
+                var regexObj = /storage-(.*)-collection/;
+                var allKeys = Object.keys(storage.data.children);
+                _.forEach(allKeys, function(key) {
+                    if (regexObj.test(key)) {
+                        _.forEach(storage.data.children[key], function(ele) {
+                            controllers.push(key.split('-')[1].toLowerCase()
```
Comment (pengz1) [inline lib/api/redfish-1.0/systems.js]: You need align with Lingling, .toLowerCase() is unnecessary.

**Comment by iceiilin** [comment]
I am OK with the logic here. And is it possible to increase the test coverage?

**Comment by mcgG** [comment]
test this please

**Comment by mcgG** [comment]
test this please

**Pull request merged**